### ISSUE TITLE
recordTest() gives error ms is not a length 1 integer

### ISSUE THREAD

**Issue by samuelhuerga** [rstudio/shinytest#160]
Hi all,

I want to use shinytest to test a shiny app I develop, but I can get it to work.
I have updated R, RStudio and the packages my app depends, and it run properly, but when running recordTest() it gives the following error:

```
> recordTest()
Error: ms is not a length 1 integer
```

I've trid to debug the function, and the point where it chrashes is at:
`app <- ShinyDriver$new(app, seed = seed, loadTimeout = 10000)`

Do I need another package?

My session info:
```
> sessionInfo()
R version 3.4.2 (2017-09-28)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252    LC_MONETARY=Spanish_Spain.1252
[4] LC_NUMERIC=C                   LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] shinytest_1.2.0.9000

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.13         rstudioapi_0.7       magrittr_1.5         debugme_1.1.0        xtable_1.8-2        
 [6] showimage_1.0.0      R6_2.2.2             rematch_1.0.1        webdriver_1.0.2.9002 httr_1.3.1          
[11] tools_3.4.2          webshot_0.4.2        parsedate_1.1.3      png_0.1-7            withr_2.1.0         
[16] htmltools_0.3.6      yaml_2.1.14          assertthat_0.2.0     digest_0.6.12        crayon_1.3.4        
[21] processx_3.0.0       shiny_1.0.5          base64enc_0.1-3      testthat_1.0.2       curl_3.0            
[26] mime_0.5             compiler_3.4.2       pingr_1.1.2          jsonlite_1.5         httpuv_1.3.5 
```

Thank you in advance!

**Comment by edavidaja** [comment]
I'm having the same issue.
```
> sessionInfo()
R version 3.4.2 (2017-09-28)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.4.2  tools_3.4.2     yaml_2.1.14     packrat_0.4.8-1
```

A couple of further bits of information:
- The installation failed initially and I had to call the `quick` option of `install_github` to get it not to check the 32 bit architecture, at which point the install succeeded (see below).
  - I did try it with the most up-to-date version of Rcpp, which did not resolve the error.
  - I had the same problem when I tried to run it out of a docker container.
```
> devtools::install_github("rstudio/shinytest")
Downloading GitHub repo rstudio/shinytest@master
from URL https://api.github.com/repos/rstudio/shinytest/zipball/master
Installing shinytest
"C:/PROGRA~1/R/R-34~1.2/bin/x64/R" --no-site-file --no-environ --no-save  \
  --no-restore --quiet CMD INSTALL  \
  "C:/Users/edavi/AppData/Local/Temp/RtmpwfyNpO/devtools2764432a3b52/rstudio-shinytest-6fe4d3f"  \
  --library=".../packrat/lib/x86_64-w64-mingw32/3.4.2"  \
  --install-tests 

* installing *source* package 'shinytest' ...
** R
** inst
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
*** arch - i386
Error: package or namespace load failed for 'shinytest' in library.dynam(lib, package, package.lib):
 DLL 'Rcpp' not found: maybe not installed for this architecture?
Error: loading failed
Execution halted
*** arch - x64
ERROR: loading failed for 'i386'
* removing '.../packrat/lib/x86_64-w64-mingw32/3.4.2/shinytest'
Installation failed: Command failed (1)
> ?devtools::install
> devtools::install_github("rstudio/shinytest", quick = TRUE)
Downloading GitHub repo rstudio/shinytest@master
from URL https://api.github.com/repos/rstudio/shinytest/zipball/master
Installing shinytest
"C:/PROGRA~1/R/R-34~1.2/bin/x64/R" --no-site-file --no-environ --no-save  \
  --no-restore --quiet CMD INSTALL  \
  "C:/Users/edavi/AppData/Local/Temp/RtmpwfyNpO/devtools27644b1150a9/rstudio-shinytest-6fe4d3f"  \
  --library=".../packrat/lib/x86_64-w64-mingw32/3.4.2"  \
  --install-tests --no-docs --no-multiarch --no-demo 

* installing *source* package 'shinytest' ...
** R
** inst
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (shinytest)
```
- `processx` gives the following warning while the 32-bit architecture is installing:
```
processx-connection.c: In function 'processx_connection_create':
processx-connection.c:74:3: warning: 'result' may be used uninitialized in this function [-Wmaybe-uninitialized]
   return result;
   ^
```

**Comment by tomasbarcellos** [comment]
This issue is also caught in [Travis' test](https://travis-ci.org/rstudio/shinytest/jobs/304812393) on `release R`.

**Comment by wch** [comment]
Sorry for the delay in replying. I've tried this out on my Mac and in a Windows VM, and have been able to reproduce the problem only occasionally.

When I try to record a test for an app for which the necessary packages are not installed, I sometimes get the `Error: ms is not a length 1 integer` for the first run (and sometimes a few more) but later I get a more informative error message.

```
> recordTest('cp')
Error: ms is not a length 1 integer
> recordTest('cp')
Error: ms is not a length 1 integer
> recordTest('cp')
Error in sd_startShiny(self, private, path, seed) : 
  Error starting application:
Running application in test mode.
Loading required package: shiny
Error in loadNamespace(name) : there is no package called 'colourpicker'
Calls: <Anonymous> ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
Execution halted
```

The same happens when I just run `ShinyDriver$new('cp')` instead of `recordTest('cp')`.


This was on Windows 32-bit (I was unable to reproduce the error on a Mac).

```
> sessioninfo::session_info()
─ Session info ───────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.4.0 (2017-04-21)
 os       Windows 10 x64              
 system   i386, mingw32               
 ui       Rgui                        
 language (EN)                        
 collate  English_United States.1252  
 tz       America/Los_Angeles         
 date     2017-11-27                  

─ Packages ───────────────────────────────────────────────────────────────────
 package     * version    date       source                                
 assertthat    0.2.0      2017-04-11 CRAN (R 3.4.0)                        
 base64enc     0.1-3      2015-07-28 CRAN (R 3.4.0)                        
 clisymbols    1.2.0      2017-05-21 CRAN (R 3.4.1)                        
 crayon        1.3.4      2017-09-16 CRAN (R 3.4.1)                        
 curl          3.0        2017-10-06 CRAN (R 3.4.2)                        
 debugme       1.1.0      2017-11-27 Github (gaborcsardi/debugme@bd89f5c)  
 digest        0.6.12     2017-01-27 CRAN (R 3.4.0)                        
 htmltools     0.3.6      2017-04-28 CRAN (R 3.4.0)                        
 httpuv        1.3.5.9002 2017-10-11 Github (rstudio/httpuv@649fc77)       
 httr          1.3.1      2017-08-20 CRAN (R 3.4.1)                        
 jsonlite      1.5.9000   2017-11-27 Github (jeroen/jsonlite@fc6c4ad)      
 later         0.6        2017-11-04 CRAN (R 3.4.2)                        
 magrittr      1.5        2014-11-22 CRAN (R 3.4.0)                        
 mime          0.5        2016-07-07 CRAN (R 3.4.0)                        
 parsedate     1.1.3      2017-03-02 CRAN (R 3.4.1)                        
 pingr         1.1.2      2017-03-02 CRAN (R 3.4.1)                        
 png           0.1-7      2013-12-03 CRAN (R 3.4.0)                        
 processx      3.0.0.9000 2017-11-27 Github (r-lib/processx@a4cccb6)       
 promises      0.1.0.9000 2017-10-11 Github (rstudio/promises@b7060b3)     
 R6            2.2.2      2017-06-17 CRAN (R 3.4.1)                        
 Rcpp          0.12.14    2017-11-23 CRAN (R 3.4.2)                        
 rematch       1.0.1      2016-04-21 CRAN (R 3.4.1)                        
 sessioninfo   1.0.0      2017-06-21 CRAN (R 3.4.1)                        
 shiny         1.0.5      2017-08-23 CRAN (R 3.4.1)                        
 shinytest   * 1.2.0.9000 2017-11-27 Github (rstudio/shinytest@0f8dce9)    
 showimage     1.0.0      2017-07-10 Github (MangoTheCat/showimage@d9000f0)
 testthat      1.0.2      2016-04-23 CRAN (R 3.4.0)                        
 webdriver     1.0.2.9002 2017-08-04 Github (rstudio/webdriver@2ccad63)    
 withr         2.1.0      2017-11-01 CRAN (R 3.4.2)                        
 xtable        1.8-2      2016-02-05 CRAN (R 3.4.0)     
```

It looks like the error is coming from processx: https://github.com/r-lib/processx/blob/a4cccb6/R/assertions.R#L37-L39

@tomasbarcellos I saw that in the Travis logs, but when I re-ran the build, it completed without any problem.

@edavidaja Thanks for pointing out the processx issue. I've filed it at r-lib/processx#84. It's possible that it is related to the main problem here.

**Comment by wch** [comment]
This should now be fixed by r-lib/processx#86. When you get a chance, please test it out!

You can install it with:

```
devtools::install_github('r-lib/processx')
```

**Comment by wch** [comment]
Also, make sure to restart R before running that command, and after. On Windows, you can't install a package that's currently loaded if it contains compiled code, as processx does. If any running R session has processx loaded, the installation will fail.
